casper/src/main/scala/coop/rchain/casper/genesis/Genesis.scala

```diff
@@ -165,52 +158,45 @@ object Genesis {
       .computeState(startHash)(blessedTerms)
       .map {
         case (stateHash, processedDeploys) =>
-          createInternalProcessedDeploy(initial, startHash, stateHash, processedDeploys)
+          createInternalProcessedDeploy(genesis, startHash, stateHash, processedDeploys)
       }
   }
 
   private def createInternalProcessedDeploy(
-      initial: BlockMessage,
+      genesis: Genesis,
       startHash: StateHash,
       stateHash: StateHash,
       processedDeploys: Seq[InternalProcessedDeploy]
   ): BlockMessage = {
-    val stateWithContracts = for {
-      bd <- initial.body
-      ps <- bd.state
-    } yield ps.withPreStateHash(startHash).withPostStateHash(stateHash)
-    val version   = initial.header.get.version
-    val timestamp = initial.header.get.timestamp
+    import genesis._
```

Comment:
Nitpick: I would prefer `genesis.proofOfStake` etc but I guess if you use this pattern enough my eyes will get used to it

Reply:
My personal preference is also `genesis.whatever`